From 260bcf5c6c47adf6ee9b5168d33f12ddc49004c6 Mon Sep 17 00:00:00 2001
From: Locus Wei-Han Chen <locus84@andestech.com>
Date: Mon, 31 Jul 2023 17:02:49 +0800
Subject: [PATCH 125/191] andes: perf: add andes event to JSON file

Add andes event to JSON file for a25 ax25 a45 and ax65

Signed-off-by: Locus Wei-Han Chen <locus84@andestech.com>
---
 .../arch/riscv/andes/A25/firmware.json        |  74 +++++
 .../arch/riscv/andes/A25/riscvstd.json        | 257 +++++++++++++++++
 .../arch/riscv/andes/A45/firmware.json        |  74 +++++
 .../arch/riscv/andes/A45/riscvstd.json        | 267 ++++++++++++++++++
 .../arch/riscv/andes/AX25/firmware.json       |  74 +++++
 .../arch/riscv/andes/AX25/riscvstd.json       | 257 +++++++++++++++++
 .../arch/riscv/andes/AX45/riscvstd.json       |  25 +-
 .../arch/riscv/andes/AX65/firmware.json       |  74 +++++
 .../arch/riscv/andes/AX65/riscvstd.json       | 262 +++++++++++++++++
 tools/perf/pmu-events/arch/riscv/mapfile.csv  |   4 +
 10 files changed, 1363 insertions(+), 5 deletions(-)
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/A25/firmware.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/A25/riscvstd.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/A45/firmware.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/A45/riscvstd.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/AX25/firmware.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/AX25/riscvstd.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/AX65/firmware.json
 create mode 100644 tools/perf/pmu-events/arch/riscv/andes/AX65/riscvstd.json

diff --git a/tools/perf/pmu-events/arch/riscv/andes/A25/firmware.json b/tools/perf/pmu-events/arch/riscv/andes/A25/firmware.json
new file mode 100644
index 000000000000..61e42c1813fa
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/A25/firmware.json
@@ -0,0 +1,74 @@
+[
+  {
+    "ArchStdEvent": "FW_MISALIGNED_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_MISALIGNED_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ILLEGAL_INSN"
+  },
+  {
+    "ArchStdEvent": "FW_SET_TIMER"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESSES"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESS_MISSES"
+  }
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/A25/riscvstd.json b/tools/perf/pmu-events/arch/riscv/andes/A25/riscvstd.json
new file mode 100644
index 000000000000..71cf1fc3f7b2
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/A25/riscvstd.json
@@ -0,0 +1,257 @@
+[
+	{
+	"EventCode": "0x10",
+	"EventName": "cycle_count",
+	"BriefDescription": "Cycle counts"
+	},
+	{
+	"EventCode": "0x20",
+	"EventName": "inst_count",
+	"BriefDescription": "Retired instruction counts"
+	},
+	{
+	"EventCode": "0x30",
+	"EventName": "int_load_inst",
+	"BriefDescription": "Integer load instructions"
+	},
+	{
+	"EventCode": "0x40",
+	"EventName": "int_store_inst",
+	"BriefDescription": "Integer store instructions"
+	},
+	{
+	"EventCode": "0x50",
+	"EventName": "atomic_mem_op",
+	"BriefDescription": "Atomic memory operation"
+	},
+	{
+	"EventCode": "0x60",
+	"EventName": "sys_inst",
+	"BriefDescription": "System instructions"
+	},
+	{
+	"EventCode": "0x70",
+	"EventName": "int_compute_inst",
+	"BriefDescription": "Integer computational instruction"
+	},
+	{
+	"EventCode": "0x80",
+	"EventName": "condition_br",
+	"BriefDescription": "Conditional branch"
+	},
+	{
+	"EventCode": "0x90",
+	"EventName": "taken_condition_br",
+	"BriefDescription": "Taken conditional branch"
+	},
+	{
+	"EventCode": "0xA0",
+	"EventName": "jal_inst",
+	"BriefDescription": "JAL instruction"
+	},
+	{
+	"EventCode": "0xB0",
+	"EventName": "jalr_inst",
+	"BriefDescription": "JALR instruction"
+	},
+	{
+	"EventCode": "0xC0",
+	"EventName": "ret_inst",
+	"BriefDescription": "Return instruction"
+	},
+	{
+	"EventCode": "0xD0",
+	"EventName": "control_trans_inst",
+	"BriefDescription": "Control transfer instruction"
+	},
+	{
+	"EventCode": "0xE0",
+	"EventName": "ex9_inst",
+	"BriefDescription": "EX9 instruction"
+	},
+	{
+	"EventCode": "0xF0",
+	"EventName": "int_mul_inst",
+	"BriefDescription": "Integer multiplication instruction"
+	},
+	{
+	"EventCode": "0x100",
+	"EventName": "int_div_rem_inst",
+	"BriefDescription": "Integer division/remainder instruction"
+	},
+	{
+	"EventCode": "0x110",
+	"EventName": "float_load_inst",
+	"BriefDescription": "Floating-point load instruction"
+	},
+	{
+	"EventCode": "0x120",
+	"EventName": "float_store_inst",
+	"BriefDescription": "Floating-point store instruction"
+	},
+	{
+	"EventCode": "0x130",
+	"EventName": "float_add_sub_inst",
+	"BriefDescription": "Floating-point addition/subtraction"
+	},
+	{
+	"EventCode": "0x140",
+	"EventName": "float_mul_inst",
+	"BriefDescription": "Floating-point multiplication"
+	},
+	{
+	"EventCode": "0x150",
+	"EventName": "float_fused_muladd_inst",
+	"BriefDescription": "Floating-point fused multiply-add"
+	},
+	{
+	"EventCode": "0x160",
+	"EventName": "float_div_sqrt_inst",
+	"BriefDescription": "Floating-point division or square-root"
+	},
+	{
+	"EventCode": "0x170",
+	"EventName": "other_float_inst",
+	"BriefDescription": "Other floating-point instruction"
+	},
+	{
+	"EventCode": "0x180",
+	"EventName": "integer_mul_and_sub_inst_count",
+	"BriefDescription": "Integer multiplication and add/sub instruction count"
+	},
+	{
+	"EventCode": "0x01",
+	"EventName": "ilm_access",
+	"BriefDescription": "ILM access"
+	},
+	{
+	"EventCode": "0x11",
+	"EventName": "dlm_access",
+	"BriefDescription": "DLM access"
+	},
+	{
+	"EventCode": "0x21",
+	"EventName": "icache_access",
+	"BriefDescription": "ICACHE access"
+	},
+	{
+	"EventCode": "0x31",
+	"EventName": "icache_miss",
+	"BriefDescription": "ICACHE miss"
+	},
+	{
+	"EventCode": "0x41",
+	"EventName": "dcache_access",
+	"BriefDescription": "DCACHE access"
+	},
+	{
+	"EventCode": "0x51",
+	"EventName": "dcache_miss",
+	"BriefDescription": "DCACHE miss"
+	},
+	{
+	"EventCode": "0x61",
+	"EventName": "dcache_load_access",
+	"BriefDescription": "DCACHE load access"
+	},
+	{
+	"EventCode": "0x71",
+	"EventName": "dcache_load_miss",
+	"BriefDescription": "DCACHE load miss"
+	},
+	{
+	"EventCode": "0x81",
+	"EventName": "dcache_store_access",
+	"BriefDescription": "DCACHE store access"
+	},
+	{
+	"EventCode": "0x91",
+	"EventName": "dcache_store_miss",
+	"BriefDescription": "DCACHE store miss"
+	},
+	{
+	"EventCode": "0xA1",
+	"EventName": "dcache_wb",
+	"BriefDescription": "DCACHE writeback"
+	},
+	{
+	"EventCode": "0xB1",
+	"EventName": "cycle_wait_icache_fill",
+	"BriefDescription": "Cycles waiting for ICACHE fill data"
+	},
+	{
+	"EventCode": "0xC1",
+	"EventName": "cycle_wait_dcache_fill",
+	"BriefDescription": "Cycles waiting for DCACHE fill data"
+	},
+	{
+	"EventCode": "0xD1",
+	"EventName": "uncached_ifetch_from_bus",
+	"BriefDescription": "Uncached ifetch data access from bus"
+	},
+	{
+	"EventCode": "0xE1",
+	"EventName": "uncached_load_from_bus",
+	"BriefDescription": "Uncached load data access from bus"
+	},
+	{
+	"EventCode": "0xF1",
+	"EventName": "cycle_wait_uncached_ifetch",
+	"BriefDescription": "Cycles waiting for uncached ifetch data from bus"
+	},
+	{
+	"EventCode": "0x101",
+	"EventName": "cycle_wait_uncached_load",
+	"BriefDescription": "Cycles waiting for uncached load data from bus"
+	},
+	{
+	"EventCode": "0x111",
+	"EventName": "main_itlb_access",
+	"BriefDescription": "Main ITLB access"
+	},
+	{
+	"EventCode": "0x121",
+	"EventName": "main_itlb_miss",
+	"BriefDescription": "Main ITLB miss"
+	},
+	{
+	"EventCode": "0x131",
+	"EventName": "main_dtlb_access",
+	"BriefDescription": "Main DTLB access"
+	},
+	{
+	"EventCode": "0x141",
+	"EventName": "main_dtlb_miss",
+	"BriefDescription": "Main DTLB miss"
+	},
+	{
+	"EventCode": "0x151",
+	"EventName": "cycle_wait_itlb_fill",
+	"BriefDescription": "Cycles waiting for Main ITLB fill data"
+	},
+	{
+	"EventCode": "0x161",
+	"EventName": "pipe_stall_cycle_dtlb_miss",
+	"BriefDescription": "Pipeline stall cycles caused by Main DTLB miss"
+	},
+	{
+	"EventCode": "0x02",
+	"EventName": "mispredict_condition_br",
+	"BriefDescription": "Misprediction of conditional branches"
+	},
+	{
+	"EventCode": "0x12",
+	"EventName": "mispredict_take_condition_br",
+	"BriefDescription": "Misprediction of taken conditional branches"
+	},
+	{
+	"EventCode": "0x22",
+	"EventName": "mispredict_target_ret_inst",
+	"BriefDescription": "Misprediction of targets of Return instructions"
+	},
+	{
+	"EventCode": "0x32",
+	"EventName": "replay_load_store",
+	"BriefDescription": "Replay for load-after-store or store-after-store cases"
+	}
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/A45/firmware.json b/tools/perf/pmu-events/arch/riscv/andes/A45/firmware.json
new file mode 100644
index 000000000000..61e42c1813fa
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/A45/firmware.json
@@ -0,0 +1,74 @@
+[
+  {
+    "ArchStdEvent": "FW_MISALIGNED_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_MISALIGNED_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ILLEGAL_INSN"
+  },
+  {
+    "ArchStdEvent": "FW_SET_TIMER"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESSES"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESS_MISSES"
+  }
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/A45/riscvstd.json b/tools/perf/pmu-events/arch/riscv/andes/A45/riscvstd.json
new file mode 100644
index 000000000000..def7e88cfc77
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/A45/riscvstd.json
@@ -0,0 +1,267 @@
+[
+	{
+	"EventCode": "0x10",
+	"EventName": "cycle_count",
+	"BriefDescription": "Cycle counts"
+	},
+	{
+	"EventCode": "0x20",
+	"EventName": "inst_count",
+	"BriefDescription": "Retired instruction counts"
+	},
+	{
+	"EventCode": "0x30",
+	"EventName": "int_load_inst",
+	"BriefDescription": "Integer load instructions"
+	},
+	{
+	"EventCode": "0x40",
+	"EventName": "int_store_inst",
+	"BriefDescription": "Integer store instructions"
+	},
+	{
+	"EventCode": "0x50",
+	"EventName": "atomic_mem_op",
+	"BriefDescription": "Atomic memory operation"
+	},
+	{
+	"EventCode": "0x60",
+	"EventName": "sys_inst",
+	"BriefDescription": "System instructions"
+	},
+	{
+	"EventCode": "0x70",
+	"EventName": "int_compute_inst",
+	"BriefDescription": "Integer computational instruction"
+	},
+	{
+	"EventCode": "0x80",
+	"EventName": "condition_br",
+	"BriefDescription": "Conditional branch"
+	},
+	{
+	"EventCode": "0x90",
+	"EventName": "taken_condition_br",
+	"BriefDescription": "Taken conditional branch"
+	},
+	{
+	"EventCode": "0xA0",
+	"EventName": "jal_inst",
+	"BriefDescription": "JAL instruction"
+	},
+	{
+	"EventCode": "0xB0",
+	"EventName": "jalr_inst",
+	"BriefDescription": "JALR instruction"
+	},
+	{
+	"EventCode": "0xC0",
+	"EventName": "ret_inst",
+	"BriefDescription": "Return instruction"
+	},
+	{
+	"EventCode": "0xD0",
+	"EventName": "control_trans_inst",
+	"BriefDescription": "Control transfer instruction"
+	},
+	{
+	"EventCode": "0xE0",
+	"EventName": "ex9_inst",
+	"BriefDescription": "EX9 instruction"
+	},
+	{
+	"EventCode": "0xF0",
+	"EventName": "int_mul_inst",
+	"BriefDescription": "Integer multiplication instruction"
+	},
+	{
+	"EventCode": "0x100",
+	"EventName": "int_div_rem_inst",
+	"BriefDescription": "Integer division/remainder instruction"
+	},
+	{
+	"EventCode": "0x110",
+	"EventName": "float_load_inst",
+	"BriefDescription": "Floating-point load instruction"
+	},
+	{
+	"EventCode": "0x120",
+	"EventName": "float_store_inst",
+	"BriefDescription": "Floating-point store instruction"
+	},
+	{
+	"EventCode": "0x130",
+	"EventName": "float_add_sub_inst",
+	"BriefDescription": "Floating-point addition/subtraction"
+	},
+	{
+	"EventCode": "0x140",
+	"EventName": "float_mul_inst",
+	"BriefDescription": "Floating-point multiplication"
+	},
+	{
+	"EventCode": "0x150",
+	"EventName": "float_fused_muladd_inst",
+	"BriefDescription": "Floating-point fused multiply-add"
+	},
+	{
+	"EventCode": "0x160",
+	"EventName": "float_div_sqrt_inst",
+	"BriefDescription": "Floating-point division or square-root"
+	},
+	{
+	"EventCode": "0x170",
+	"EventName": "other_float_inst",
+	"BriefDescription": "Other floating-point instruction"
+	},
+	{
+	"EventCode": "0x180",
+	"EventName": "integer_mul_and_sub_inst_count",
+	"BriefDescription": "Integer multiplication and add/sub instruction count"
+	},
+	{
+	"EventCode": "0x190",
+	"EventName": "op_count",
+	"BriefDescription": "Retired operation count"
+	},
+	{
+	"EventCode": "0x01",
+	"EventName": "ilm_access",
+	"BriefDescription": "ILM access"
+	},
+	{
+	"EventCode": "0x11",
+	"EventName": "dlm_access",
+	"BriefDescription": "DLM access"
+	},
+	{
+	"EventCode": "0x21",
+	"EventName": "icache_access",
+	"BriefDescription": "ICACHE access"
+	},
+	{
+	"EventCode": "0x31",
+	"EventName": "icache_miss",
+	"BriefDescription": "ICACHE miss"
+	},
+	{
+	"EventCode": "0x41",
+	"EventName": "dcache_access",
+	"BriefDescription": "DCACHE access"
+	},
+	{
+	"EventCode": "0x51",
+	"EventName": "dcache_miss",
+	"BriefDescription": "DCACHE miss"
+	},
+	{
+	"EventCode": "0x61",
+	"EventName": "dcache_load_access",
+	"BriefDescription": "DCACHE load access"
+	},
+	{
+	"EventCode": "0x71",
+	"EventName": "dcache_load_miss",
+	"BriefDescription": "DCACHE load miss"
+	},
+	{
+	"EventCode": "0x81",
+	"EventName": "dcache_store_access",
+	"BriefDescription": "DCACHE store access"
+	},
+	{
+	"EventCode": "0x91",
+	"EventName": "dcache_store_miss",
+	"BriefDescription": "DCACHE store miss"
+	},
+	{
+	"EventCode": "0xA1",
+	"EventName": "dcache_wb",
+	"BriefDescription": "DCACHE writeback"
+	},
+	{
+	"EventCode": "0xB1",
+	"EventName": "cycle_wait_icache_fill",
+	"BriefDescription": "Cycles waiting for ICACHE fill data"
+	},
+	{
+	"EventCode": "0xC1",
+	"EventName": "cycle_wait_dcache_fill",
+	"BriefDescription": "Cycles waiting for DCACHE fill data"
+	},
+	{
+	"EventCode": "0xD1",
+	"EventName": "uncached_ifetch_from_bus",
+	"BriefDescription": "Uncached ifetch data access from bus"
+	},
+	{
+	"EventCode": "0xE1",
+	"EventName": "uncached_load_from_bus",
+	"BriefDescription": "Uncached load data access from bus"
+	},
+	{
+	"EventCode": "0xF1",
+	"EventName": "cycle_wait_uncached_ifetch",
+	"BriefDescription": "Cycles waiting for uncached ifetch data from bus"
+	},
+	{
+	"EventCode": "0x101",
+	"EventName": "cycle_wait_uncached_load",
+	"BriefDescription": "Cycles waiting for uncached load data from bus"
+	},
+	{
+	"EventCode": "0x111",
+	"EventName": "main_itlb_access",
+	"BriefDescription": "Main ITLB access"
+	},
+	{
+	"EventCode": "0x121",
+	"EventName": "main_itlb_miss",
+	"BriefDescription": "Main ITLB miss"
+	},
+	{
+	"EventCode": "0x131",
+	"EventName": "main_dtlb_access",
+	"BriefDescription": "Main DTLB access"
+	},
+	{
+	"EventCode": "0x141",
+	"EventName": "main_dtlb_miss",
+	"BriefDescription": "Main DTLB miss"
+	},
+	{
+	"EventCode": "0x151",
+	"EventName": "cycle_wait_itlb_fill",
+	"BriefDescription": "Cycles waiting for Main ITLB fill data"
+	},
+	{
+	"EventCode": "0x161",
+	"EventName": "pipe_stall_cycle_dtlb_miss",
+	"BriefDescription": "Pipeline stall cycles caused by Main DTLB miss"
+	},
+	{
+	"EventCode": "0x171",
+	"EventName": "prefetch_bus_access",
+	"BriefDescription": "Hardware prefetch bus access"
+	},
+	{
+	"EventCode": "0x181",
+	"EventName": "cycle_wait_for_operand",
+	"BriefDescription": "Cycles waiting for source operand ready in the integer register file scoreboard"
+	},
+	{
+	"EventCode": "0x02",
+	"EventName": "mispredict_condition_br",
+	"BriefDescription": "Misprediction of conditional branches"
+	},
+	{
+	"EventCode": "0x12",
+	"EventName": "mispredict_take_condition_br",
+	"BriefDescription": "Misprediction of taken conditional branches"
+	},
+	{
+	"EventCode": "0x22",
+	"EventName": "mispredict_target_ret_inst",
+	"BriefDescription": "Misprediction of targets of Return instructions"
+	}
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/AX25/firmware.json b/tools/perf/pmu-events/arch/riscv/andes/AX25/firmware.json
new file mode 100644
index 000000000000..61e42c1813fa
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/AX25/firmware.json
@@ -0,0 +1,74 @@
+[
+  {
+    "ArchStdEvent": "FW_MISALIGNED_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_MISALIGNED_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ILLEGAL_INSN"
+  },
+  {
+    "ArchStdEvent": "FW_SET_TIMER"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESSES"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESS_MISSES"
+  }
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/AX25/riscvstd.json b/tools/perf/pmu-events/arch/riscv/andes/AX25/riscvstd.json
new file mode 100644
index 000000000000..71cf1fc3f7b2
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/AX25/riscvstd.json
@@ -0,0 +1,257 @@
+[
+	{
+	"EventCode": "0x10",
+	"EventName": "cycle_count",
+	"BriefDescription": "Cycle counts"
+	},
+	{
+	"EventCode": "0x20",
+	"EventName": "inst_count",
+	"BriefDescription": "Retired instruction counts"
+	},
+	{
+	"EventCode": "0x30",
+	"EventName": "int_load_inst",
+	"BriefDescription": "Integer load instructions"
+	},
+	{
+	"EventCode": "0x40",
+	"EventName": "int_store_inst",
+	"BriefDescription": "Integer store instructions"
+	},
+	{
+	"EventCode": "0x50",
+	"EventName": "atomic_mem_op",
+	"BriefDescription": "Atomic memory operation"
+	},
+	{
+	"EventCode": "0x60",
+	"EventName": "sys_inst",
+	"BriefDescription": "System instructions"
+	},
+	{
+	"EventCode": "0x70",
+	"EventName": "int_compute_inst",
+	"BriefDescription": "Integer computational instruction"
+	},
+	{
+	"EventCode": "0x80",
+	"EventName": "condition_br",
+	"BriefDescription": "Conditional branch"
+	},
+	{
+	"EventCode": "0x90",
+	"EventName": "taken_condition_br",
+	"BriefDescription": "Taken conditional branch"
+	},
+	{
+	"EventCode": "0xA0",
+	"EventName": "jal_inst",
+	"BriefDescription": "JAL instruction"
+	},
+	{
+	"EventCode": "0xB0",
+	"EventName": "jalr_inst",
+	"BriefDescription": "JALR instruction"
+	},
+	{
+	"EventCode": "0xC0",
+	"EventName": "ret_inst",
+	"BriefDescription": "Return instruction"
+	},
+	{
+	"EventCode": "0xD0",
+	"EventName": "control_trans_inst",
+	"BriefDescription": "Control transfer instruction"
+	},
+	{
+	"EventCode": "0xE0",
+	"EventName": "ex9_inst",
+	"BriefDescription": "EX9 instruction"
+	},
+	{
+	"EventCode": "0xF0",
+	"EventName": "int_mul_inst",
+	"BriefDescription": "Integer multiplication instruction"
+	},
+	{
+	"EventCode": "0x100",
+	"EventName": "int_div_rem_inst",
+	"BriefDescription": "Integer division/remainder instruction"
+	},
+	{
+	"EventCode": "0x110",
+	"EventName": "float_load_inst",
+	"BriefDescription": "Floating-point load instruction"
+	},
+	{
+	"EventCode": "0x120",
+	"EventName": "float_store_inst",
+	"BriefDescription": "Floating-point store instruction"
+	},
+	{
+	"EventCode": "0x130",
+	"EventName": "float_add_sub_inst",
+	"BriefDescription": "Floating-point addition/subtraction"
+	},
+	{
+	"EventCode": "0x140",
+	"EventName": "float_mul_inst",
+	"BriefDescription": "Floating-point multiplication"
+	},
+	{
+	"EventCode": "0x150",
+	"EventName": "float_fused_muladd_inst",
+	"BriefDescription": "Floating-point fused multiply-add"
+	},
+	{
+	"EventCode": "0x160",
+	"EventName": "float_div_sqrt_inst",
+	"BriefDescription": "Floating-point division or square-root"
+	},
+	{
+	"EventCode": "0x170",
+	"EventName": "other_float_inst",
+	"BriefDescription": "Other floating-point instruction"
+	},
+	{
+	"EventCode": "0x180",
+	"EventName": "integer_mul_and_sub_inst_count",
+	"BriefDescription": "Integer multiplication and add/sub instruction count"
+	},
+	{
+	"EventCode": "0x01",
+	"EventName": "ilm_access",
+	"BriefDescription": "ILM access"
+	},
+	{
+	"EventCode": "0x11",
+	"EventName": "dlm_access",
+	"BriefDescription": "DLM access"
+	},
+	{
+	"EventCode": "0x21",
+	"EventName": "icache_access",
+	"BriefDescription": "ICACHE access"
+	},
+	{
+	"EventCode": "0x31",
+	"EventName": "icache_miss",
+	"BriefDescription": "ICACHE miss"
+	},
+	{
+	"EventCode": "0x41",
+	"EventName": "dcache_access",
+	"BriefDescription": "DCACHE access"
+	},
+	{
+	"EventCode": "0x51",
+	"EventName": "dcache_miss",
+	"BriefDescription": "DCACHE miss"
+	},
+	{
+	"EventCode": "0x61",
+	"EventName": "dcache_load_access",
+	"BriefDescription": "DCACHE load access"
+	},
+	{
+	"EventCode": "0x71",
+	"EventName": "dcache_load_miss",
+	"BriefDescription": "DCACHE load miss"
+	},
+	{
+	"EventCode": "0x81",
+	"EventName": "dcache_store_access",
+	"BriefDescription": "DCACHE store access"
+	},
+	{
+	"EventCode": "0x91",
+	"EventName": "dcache_store_miss",
+	"BriefDescription": "DCACHE store miss"
+	},
+	{
+	"EventCode": "0xA1",
+	"EventName": "dcache_wb",
+	"BriefDescription": "DCACHE writeback"
+	},
+	{
+	"EventCode": "0xB1",
+	"EventName": "cycle_wait_icache_fill",
+	"BriefDescription": "Cycles waiting for ICACHE fill data"
+	},
+	{
+	"EventCode": "0xC1",
+	"EventName": "cycle_wait_dcache_fill",
+	"BriefDescription": "Cycles waiting for DCACHE fill data"
+	},
+	{
+	"EventCode": "0xD1",
+	"EventName": "uncached_ifetch_from_bus",
+	"BriefDescription": "Uncached ifetch data access from bus"
+	},
+	{
+	"EventCode": "0xE1",
+	"EventName": "uncached_load_from_bus",
+	"BriefDescription": "Uncached load data access from bus"
+	},
+	{
+	"EventCode": "0xF1",
+	"EventName": "cycle_wait_uncached_ifetch",
+	"BriefDescription": "Cycles waiting for uncached ifetch data from bus"
+	},
+	{
+	"EventCode": "0x101",
+	"EventName": "cycle_wait_uncached_load",
+	"BriefDescription": "Cycles waiting for uncached load data from bus"
+	},
+	{
+	"EventCode": "0x111",
+	"EventName": "main_itlb_access",
+	"BriefDescription": "Main ITLB access"
+	},
+	{
+	"EventCode": "0x121",
+	"EventName": "main_itlb_miss",
+	"BriefDescription": "Main ITLB miss"
+	},
+	{
+	"EventCode": "0x131",
+	"EventName": "main_dtlb_access",
+	"BriefDescription": "Main DTLB access"
+	},
+	{
+	"EventCode": "0x141",
+	"EventName": "main_dtlb_miss",
+	"BriefDescription": "Main DTLB miss"
+	},
+	{
+	"EventCode": "0x151",
+	"EventName": "cycle_wait_itlb_fill",
+	"BriefDescription": "Cycles waiting for Main ITLB fill data"
+	},
+	{
+	"EventCode": "0x161",
+	"EventName": "pipe_stall_cycle_dtlb_miss",
+	"BriefDescription": "Pipeline stall cycles caused by Main DTLB miss"
+	},
+	{
+	"EventCode": "0x02",
+	"EventName": "mispredict_condition_br",
+	"BriefDescription": "Misprediction of conditional branches"
+	},
+	{
+	"EventCode": "0x12",
+	"EventName": "mispredict_take_condition_br",
+	"BriefDescription": "Misprediction of taken conditional branches"
+	},
+	{
+	"EventCode": "0x22",
+	"EventName": "mispredict_target_ret_inst",
+	"BriefDescription": "Misprediction of targets of Return instructions"
+	},
+	{
+	"EventCode": "0x32",
+	"EventName": "replay_load_store",
+	"BriefDescription": "Replay for load-after-store or store-after-store cases"
+	}
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/AX45/riscvstd.json b/tools/perf/pmu-events/arch/riscv/andes/AX45/riscvstd.json
index 55cd56d9e527..def7e88cfc77 100644
--- a/tools/perf/pmu-events/arch/riscv/andes/AX45/riscvstd.json
+++ b/tools/perf/pmu-events/arch/riscv/andes/AX45/riscvstd.json
@@ -115,6 +115,16 @@
 	"BriefDescription": "Other floating-point instruction"
 	},
 	{
+	"EventCode": "0x180",
+	"EventName": "integer_mul_and_sub_inst_count",
+	"BriefDescription": "Integer multiplication and add/sub instruction count"
+	},
+	{
+	"EventCode": "0x190",
+	"EventName": "op_count",
+	"BriefDescription": "Retired operation count"
+	},
+	{
 	"EventCode": "0x01",
 	"EventName": "ilm_access",
 	"BriefDescription": "ILM access"
@@ -230,6 +240,16 @@
 	"BriefDescription": "Pipeline stall cycles caused by Main DTLB miss"
 	},
 	{
+	"EventCode": "0x171",
+	"EventName": "prefetch_bus_access",
+	"BriefDescription": "Hardware prefetch bus access"
+	},
+	{
+	"EventCode": "0x181",
+	"EventName": "cycle_wait_for_operand",
+	"BriefDescription": "Cycles waiting for source operand ready in the integer register file scoreboard"
+	},
+	{
 	"EventCode": "0x02",
 	"EventName": "mispredict_condition_br",
 	"BriefDescription": "Misprediction of conditional branches"
@@ -243,10 +263,5 @@
 	"EventCode": "0x22",
 	"EventName": "mispredict_target_ret_inst",
 	"BriefDescription": "Misprediction of targets of Return instructions"
-	},
-	{
-	"EventCode": "0x32",
-	"EventName": "replay_las_sas",
-	"BriefDescription": "Replay for load-after-store or store-after-store cases"
 	}
 ]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/AX65/firmware.json b/tools/perf/pmu-events/arch/riscv/andes/AX65/firmware.json
new file mode 100644
index 000000000000..61e42c1813fa
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/AX65/firmware.json
@@ -0,0 +1,74 @@
+[
+  {
+    "ArchStdEvent": "FW_MISALIGNED_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_MISALIGNED_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_LOAD"
+  },
+  {
+    "ArchStdEvent": "FW_ACCESS_STORE"
+  },
+  {
+    "ArchStdEvent": "FW_ILLEGAL_INSN"
+  },
+  {
+    "ArchStdEvent": "FW_SET_TIMER"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_IPI_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_FENCE_I_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_SFENCE_VMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_GVMA_VMID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_SENT"
+  },
+  {
+    "ArchStdEvent": "FW_HFENCE_VVMA_ASID_RECEIVED"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESSES"
+  },
+  {
+    "ArchStdEvent": "ANDES_L2C_ACCESS_MISSES"
+  }
+]
diff --git a/tools/perf/pmu-events/arch/riscv/andes/AX65/riscvstd.json b/tools/perf/pmu-events/arch/riscv/andes/AX65/riscvstd.json
new file mode 100644
index 000000000000..916712a2e123
--- /dev/null
+++ b/tools/perf/pmu-events/arch/riscv/andes/AX65/riscvstd.json
@@ -0,0 +1,262 @@
+[
+	{
+	"EventCode": "0x10",
+	"EventName": "cycle_count",
+	"BriefDescription": "Cycle counts"
+	},
+	{
+	"EventCode": "0x20",
+	"EventName": "inst_count",
+	"BriefDescription": "Retired instruction counts"
+	},
+	{
+	"EventCode": "0x30",
+	"EventName": "int_load_inst",
+	"BriefDescription": "Integer load instructions"
+	},
+	{
+	"EventCode": "0x40",
+	"EventName": "int_store_inst",
+	"BriefDescription": "Integer store instructions"
+	},
+	{
+	"EventCode": "0x50",
+	"EventName": "atomic_mem_op",
+	"BriefDescription": "Atomic memory operation"
+	},
+	{
+	"EventCode": "0x60",
+	"EventName": "sys_inst",
+	"BriefDescription": "System instructions"
+	},
+	{
+	"EventCode": "0x70",
+	"EventName": "int_compute_inst",
+	"BriefDescription": "Integer computational instruction"
+	},
+	{
+	"EventCode": "0x80",
+	"EventName": "condition_br",
+	"BriefDescription": "Conditional branch"
+	},
+	{
+	"EventCode": "0x90",
+	"EventName": "taken_condition_br",
+	"BriefDescription": "Taken conditional branch"
+	},
+	{
+	"EventCode": "0xA0",
+	"EventName": "jal_inst",
+	"BriefDescription": "JAL instruction"
+	},
+	{
+	"EventCode": "0xB0",
+	"EventName": "jalr_inst",
+	"BriefDescription": "JALR instruction"
+	},
+	{
+	"EventCode": "0xC0",
+	"EventName": "ret_inst",
+	"BriefDescription": "Return instruction"
+	},
+	{
+	"EventCode": "0xD0",
+	"EventName": "control_trans_inst",
+	"BriefDescription": "Control transfer instruction"
+	},
+	{
+	"EventCode": "0xF0",
+	"EventName": "int_mul_inst",
+	"BriefDescription": "Integer multiplication instruction"
+	},
+	{
+	"EventCode": "0x100",
+	"EventName": "int_div_rem_inst",
+	"BriefDescription": "Integer division/remainder instruction"
+	},
+	{
+	"EventCode": "0x110",
+	"EventName": "float_load_inst",
+	"BriefDescription": "Floating-point load instruction"
+	},
+	{
+	"EventCode": "0x120",
+	"EventName": "float_store_inst",
+	"BriefDescription": "Floating-point store instruction"
+	},
+	{
+	"EventCode": "0x130",
+	"EventName": "float_add_sub_inst",
+	"BriefDescription": "Floating-point addition/subtraction"
+	},
+	{
+	"EventCode": "0x140",
+	"EventName": "float_mul_inst",
+	"BriefDescription": "Floating-point multiplication"
+	},
+	{
+	"EventCode": "0x150",
+	"EventName": "float_fused_muladd_inst",
+	"BriefDescription": "Floating-point fused multiply-add"
+	},
+	{
+	"EventCode": "0x160",
+	"EventName": "float_div_sqrt_inst",
+	"BriefDescription": "Floating-point division or square-root"
+	},
+	{
+	"EventCode": "0x170",
+	"EventName": "other_float_inst",
+	"BriefDescription": "Other floating-point instruction"
+	},
+	{
+	"EventCode": "0x190",
+	"EventName": "op_count",
+	"BriefDescription": "Retired operation count"
+	},
+	{
+	"EventCode": "0x21",
+	"EventName": "icache_access",
+	"BriefDescription": "ICACHE access"
+	},
+	{
+	"EventCode": "0x31",
+	"EventName": "icache_miss",
+	"BriefDescription": "ICACHE miss"
+	},
+	{
+	"EventCode": "0x41",
+	"EventName": "dcache_access",
+	"BriefDescription": "DCACHE access"
+	},
+	{
+	"EventCode": "0x51",
+	"EventName": "dcache_miss",
+	"BriefDescription": "DCACHE miss"
+	},
+	{
+	"EventCode": "0x61",
+	"EventName": "dcache_load_access",
+	"BriefDescription": "DCACHE load access"
+	},
+	{
+	"EventCode": "0x71",
+	"EventName": "dcache_load_miss",
+	"BriefDescription": "DCACHE load miss"
+	},
+	{
+	"EventCode": "0x81",
+	"EventName": "dcache_store_access",
+	"BriefDescription": "DCACHE store access"
+	},
+	{
+	"EventCode": "0x91",
+	"EventName": "dcache_store_miss",
+	"BriefDescription": "DCACHE store miss"
+	},
+	{
+	"EventCode": "0xA1",
+	"EventName": "dcache_wb",
+	"BriefDescription": "DCACHE writeback"
+	},
+	{
+	"EventCode": "0xB1",
+	"EventName": "cycle_wait_icache_fill",
+	"BriefDescription": "Cycles waiting for ICACHE fill data"
+	},
+	{
+	"EventCode": "0xC1",
+	"EventName": "cycle_wait_dcache_fill",
+	"BriefDescription": "Cycles waiting for DCACHE fill data"
+	},
+	{
+	"EventCode": "0xD1",
+	"EventName": "uncached_ifetch_from_bus",
+	"BriefDescription": "Uncached ifetch data access from bus"
+	},
+	{
+	"EventCode": "0xE1",
+	"EventName": "uncached_load_from_bus",
+	"BriefDescription": "Uncached load data access from bus"
+	},
+	{
+	"EventCode": "0xF1",
+	"EventName": "cycle_wait_uncached_ifetch",
+	"BriefDescription": "Cycles waiting for uncached ifetch data from bus"
+	},
+	{
+	"EventCode": "0x101",
+	"EventName": "cycle_wait_uncached_load",
+	"BriefDescription": "Cycles waiting for uncached load data from bus"
+	},
+	{
+	"EventCode": "0x111",
+	"EventName": "main_itlb_access",
+	"BriefDescription": "Main ITLB access"
+	},
+	{
+	"EventCode": "0x121",
+	"EventName": "main_itlb_miss",
+	"BriefDescription": "Main ITLB miss"
+	},
+	{
+	"EventCode": "0x131",
+	"EventName": "main_dtlb_access",
+	"BriefDescription": "Main DTLB access"
+	},
+	{
+	"EventCode": "0x141",
+	"EventName": "main_dtlb_miss",
+	"BriefDescription": "Main DTLB miss"
+	},
+	{
+	"EventCode": "0x151",
+	"EventName": "cycle_wait_itlb_fill",
+	"BriefDescription": "Cycles waiting for Main ITLB fill data"
+	},
+	{
+	"EventCode": "0x171",
+	"EventName": "prefetch_bus_access",
+	"BriefDescription": "Hardware prefetch bus access"
+	},
+	{
+	"EventCode": "0x191",
+	"EventName": "l1_itlb_access",
+	"BriefDescription": "L1 ITLB access"
+	},
+	{
+	"EventCode": "0x1A1",
+	"EventName": "l1_itlb_miss",
+	"BriefDescription": "L1 ITLB miss"
+	},
+	{
+	"EventCode": "0x1B1",
+	"EventName": "l1_dtlb_access",
+	"BriefDescription": "L1 DTLB access"
+	},
+	{
+	"EventCode": "0x1C1",
+	"EventName": "l1_dtlb_miss",
+	"BriefDescription": "L1 DTLB miss"
+	},
+	{
+	"EventCode": "0x02",
+	"EventName": "mispredict_condition_br",
+	"BriefDescription": "Misprediction of conditional branches"
+	},
+	{
+	"EventCode": "0x12",
+	"EventName": "mispredict_take_condition_br",
+	"BriefDescription": "Misprediction of taken conditional branches"
+	},
+	{
+	"EventCode": "0x22",
+	"EventName": "mispredict_target_ret_inst",
+	"BriefDescription": "Misprediction of targets of Return instructions"
+	},
+	{
+	"EventCode": "0x32",
+	"EventName": "replay_las_sas",
+	"BriefDescription": "Replay for load-after-store or store-after-store cases"
+	}
+]
diff --git a/tools/perf/pmu-events/arch/riscv/mapfile.csv b/tools/perf/pmu-events/arch/riscv/mapfile.csv
index 5f0e9a8b08ed..631257eefbeb 100644
--- a/tools/perf/pmu-events/arch/riscv/mapfile.csv
+++ b/tools/perf/pmu-events/arch/riscv/mapfile.csv
@@ -16,3 +16,7 @@
 #MVENDORID-MARCHID-MIMPID,Version,Filename,EventType
 0x489-0x8000000000000007-0x[[:xdigit:]]+,v1,sifive/u74,core
 0x31e-0x8000000000008a45-0x[[:xdigit:]]+,V5,andes/AX45,core
+0x31e-0x8000000000000a45-0x[[:xdigit:]]+,V5,andes/A45,core
+0x31e-0x8000000000008a25-0x[[:xdigit:]]+,V5,andes/AX25,core
+0x31e-0x8000000000000a25-0x[[:xdigit:]]+,V5,andes/A25,core
+0x31e-0x8000000000008a65-0x[[:xdigit:]]+,V5,andes/AX65,core
-- 
2.34.1

